src/Evolution/Systems/Cce/AnalyticSolutions/TeukolskyWave.hpp

```diff
@@ -0,0 +1,373 @@
+   * where \f$r\f$ is the extraction radius, \f$u\f$ the retarded time, \f$a\f$
+   * the amplitude of the pulse and \f$k\f$ the duration of the pulse.
+   */
+  double dr_pulse_profile_coefficient_b(double time) const noexcept;
+
+  /*
+   * The radial derivative of the C coefficient is calculated as
+   *
+   * \f[
+   * \partial_r C + \partial_t C  = -(1/4) (105 * k^8 + 16 r^4 u^4
+   * - 16 k^2 r^3 u^2 (3 r + 2 u) - 6 k^6 r (9 r + 28 u)
+   * + 12 k^4 r^2 (r^2 + 4 r u + 9 u^2))
+   *  a e^{-u^2/k^2} / (k^8 r^6)
+   * \f]
+   *
+   * where \f$r\f$ is the extraction radius, \f$u\f$ the retarded time, \f$a\f$
+   * the amplitude of the pulse and \f$k\f$ the duration of the pulse.
+   */
+  double dr_pulse_profile_coefficient_c(double time) const noexcept;
+
+  static DataVector sin_theta(size_t l_max) noexcept;
+
+  static DataVector cos_theta(size_t l_max) noexcept;
+
+ protected:
+  /// A no-op as the gauge wave solution does not have substantial
+  /// shared computation to prepare before the separate component calculations.
+  void prepare_solution(const size_t /*output_l_max*/,
+                        const double /*time*/) const noexcept override {}
+
+  /*!
+   * \brief Compute the spherical coordinate metric from the closed-form
+   * perturbative Teukolsky wave metric.
+   *
+   * \details The specific outgoing wave selected in this analytic solution is
+   * constructed from a (2,0) mode as in \cite Barkett2019uae, and takes the
+   * form
+   *
+   * \f{align*}{
+   * g_{tt} &= -1\\
+   * g_{rr} &= (1 + f_{rr}) \\
+   * g_{r \theta} &= 2 B f_{r \theta} r\\
+   * g_{\theta \theta} &= (1 + C f_{\theta \theta}^{(C)}
+   * + A f_{\theta \theta}^{(A)}) r^2\\
+   * g_{\phi \phi} &= (1 + C f_{\phi \phi}^{(C)}
+   * + A f_{\phi \phi}^{(A)}) r^2 \sin^2 \theta\\
+   * \f}
+   *
+   * and all other components vanish. The angular factors generated by the
+   * choice of spin-weighted spherical harmonic are
+   *
+   * \f{align*}{
+   * f_{rr} &= 2 - 3 \sin^2 \theta \\
+   * f_{r \theta} &=  -3 \sin \theta \cos \theta \\
+   * f_{\theta \theta}^{(C)} &= 3 \sin^2 \theta \\
+   * f_{\theta \theta}^{(A)} &= -1 \\
+   * f_{\phi \phi}^{(C)} &= - 3 \sin^2 \theta \\
+   * f_{\phi \phi}^{(A)} &= 3 \sin^2 \theta -1,
+   * \f}
+   *
+   * the radial and time dependent factors are
+   *
+   * \f{align*}{
+   * A &= 3 \left(\frac{\partial_u^2 F(u)}{r^3}
+   * + \frac{3 \partial_u F(u)}{r^4} + \frac{3 F(u)}{r^5} \right),\\
+   * B &= - \left(\frac{\partial_u^3 F(u)}{r^2}
+   * + \frac{3 \partial_u^2 F(u)}{r^3} + \frac{6 \partial_uF(u)}{r^4}
+   * + \frac{6 F(u)}{r^5}\right), \\
+   * C &= \frac{1}{4} \left(\frac{\partial_u^4 F(u)}{r}
+   * + \frac{2 \partial_u^3 F(u)}{r^2} + \frac{9 \partial_u^2 F(u)}{r^3}
+   * + \frac{21 \partial_u F(u)}{r^4} + \frac{21 F(u)}{r}\right),
+   * \f}
+   *
+   * and the pulse profile is
+   *
+   * \f[
+   * F(u) = a e^{-u^2 /k^2}.
+   * \f]
+   *
+   * \note The \f$\phi\f$ components are returned in 'pfaffian' form, where the
```

Comment:
I would suggest omitting Pfaffian in the documentation because many people don't know what it is.  Just say that we are coding up g_phi_phi with the appropriate power of sin(theta) as these factors typically cancel when multiplied by Jacobians that also are expressed with similarly written terms